### DISCUSSION TITLE
Input features in the classifier

### DISCUSSION THREAD

**Discussion by Adelakun1999** [mrdbourke/pytorch-deep-learning#945] [Q&A]
self.classifier = nn.Sequential(
        nn.Flatten(),
        nn.Linear(in_features=hidden_units*7*7,
                  out_features= output_shape)
    )

I don't seem to understand how we got the 7*7 in the code above

**Comment by kRiShNa-429407** [comment]
Please tell me in detail what you are really asking.  Also mention video number and timestamp.

**Comment by LuluW8071** [comment]
@Adelakun1999 
```python
self.classifier = nn.Sequential(
nn.Flatten(),
nn.Linear(in_features=hidden_units*7*7,
out_features= output_shape)
)
```

The job of the `nn.Flatten()` layer is to convert 2D images (height and width arrays) into 1D feature arrays. The features flattened are of size `hidden_units * 7 * 7`. If you use a `print(x.shape)` statement in your forward pass of the model before passing through `self.classifier`, you will get the correct `input_features`.

Additionally, if it doesn't match the `in_features`, it will raise a shape mismatch error, providing you with a matrix error where the correct `in_features` will be displayed.
